libethereum/State.h

```diff
@@ -98,10 +98,68 @@ DEV_SIMPLE_EXCEPTION(IncorrectAccountStartNonceInState);
 
 class SealEngineFace;
 
+
+
+namespace detail
+{
+
+/// An atomic state changelog entry.
+struct Change
+{
+	enum Kind: int
+	{
+		/// Account balance changed. Change::value contains the amount the
+		/// balance was increased by.
+		Balance,
+
+		/// Account storage was modified. Change::key contains the storage key,
+		/// Change::value the storage value.
+		Storage,
+
+		/// Account nonce was increased by 1.
+		Nonce,
+
+		/// Account was created (it was not existing before).
+		Create,
+
+		/// New code was added to an account (by "create" message execution).
+		NewCode,
+
+		/// Account was touched for the first time.
+		Touch
+	};
+
+	Kind kind;        ///< The kind of the change.
```

Comment:
I considered this. To achieve this we would need to use some polymorphic interfaces what means heap allocation and additional dereference (not CPU cache friendly). I can benchmark this, but as a separated task.

What we can do at low cost is to use `h256` (32 bytes) instead of `u256` (40 bytes).